gdal/frmts/postgisraster/postgisrasterdataset.cpp

```diff
@@ -2786,24 +2787,26 @@ GetConnectionInfo(const char * pszFilename,
     }
 
     /**
-     * Case 1: There's no database name: Error, you need, at least,
-     * specify a database name (NOTE: insensitive search)
+     * Case 1: There's no database or service name: Error, you need, at least,
+     * specify a database or a service name (NOTE: insensitive search)
      **/
     nPos = CSLFindName(papszParams, "dbname");
-    if (nPos == -1) {
+    sPos = CSLFindName(papszParams, "service");
+
+    if (nPos == -1 && sPos == -1) {
         CPLError(CE_Failure, CPLE_AppDefined,
-                "You must specify at least a db name");
+                "You must specify at least a db name or a service name");
 
         CSLDestroy(papszParams);
 
         return false;
-    }
-
-    *ppszDbname =
-        CPLStrdup(CPLParseNameValue(papszParams[nPos], nullptr));
+    } 
```

Comment:
extra white space added at that line and following one

Reply:
Fixed.